src/librustc_borrowck/borrowck/gather_loans/mod.rs

```diff
@@ -343,35 +372,25 @@ impl<'a, 'tcx> GatherLoanCtxt<'a, 'tcx> {
         // Create the loan record (if needed).
         let loan = match restr {
             RestrictionResult::Safe => {
-                // No restrictions---no loan record necessary
+                let loan_scope = loan_scope_from_region(loan_region);
+                debug!("loan_scope = {:?}", loan_scope);
+
+                let borrow_scope = region::CodeExtent::Misc(borrow_id);
+                let loan_scope = self.compute_gen_scope(borrow_scope, loan_scope);
+
+                let safe_loan = SafeLoan {
+                    kind: req_kind,
+                    loan_scope: loan_scope,
+                    span: borrow_span,
+                };
+                self.safe_loans.push(safe_loan);
```

Comment:
I wouldn't worry too much about that. Save-analysis isn't really the direction we want things to go anyway -- that is, having a distinct "save-analysis mode". I think we hope that eventually the RLS will just be polling the results of regular queries.